src/operator/contrib/two_bit_quantize.cc

```diff
@@ -0,0 +1,122 @@
+ */
+#include "./two_bit_quantize-inl.h"
+
+namespace mxnet {
+namespace op {
+
+DMLC_REGISTER_PARAMETER(TwoBitParam);
+
+NNVM_REGISTER_OP(_contrib_quantize_2bit)
+.describe(R"code(Quantize an input tensor into using 2bits for each value using
+user-specified thresholds, while storing quantization error in residual array.
+
+The quantize_2bit operator takes 5 arguments and is called as follows:
+`quantize_2bit(array, residual, out, neg_threshold, pos_threshold)`.
+The operator modifies `residual` and `out` arrays.
+The `out`variable will be the quantized array. Note that, `out` array can be generated by
+invoking `create_2bit(array)`, avoiding calculation of size of quantized array.
+This `out` array has first three elements as negative threshold, positive threshold,
+and size of the original uncompressed array. Any elements after these three elements
+represent quantized data.
+The operation sums up array and residual, and then
+applies the thresholds to quantize the data into one of three states
+represented by 2bits. 16 such quantized floats in the original array
+are packed together into one float in the `out` array.
+The quantization error is stored in residual array.
+
+For example, assume the input array (gradient) is [5.0, -1.0, -5.0, -4.0], and the
+residual is [0.0, -2.0, 0, 1.0]. Let the negative and positive thresholds be
+-4.0 and +4.0, respectively. In this method, the elements whose
+(gradient + residual) >= pos_threshold will be quantized into 2-bits '01',
+and the elements whose (gradient + residual) <= neg_threshold will be
+quantized into 2-bits '10'. The other elements will be quantized
+as '00'. Every 16 floats in the original array will be packed
+into one float variable in the output array.
+
+In this example, 'out' has 4 elements. The first element stores the
+neg_threshold (-4.0), the second element stores the pos_threshold (+4.0), the
+third element stores the original size of the uncompressed array, and the
+original array will be quantized into a single element in the last element.
+The residual is also updated to [1.0, -3.0, -1.0, -3.0].
+)code" ADD_FILELINE)
+.set_num_inputs(3)
+.set_num_outputs(0)
+.set_attr_parser(ParamParser<TwoBitParam>)
+.set_attr<nnvm::FInferShape>("FInferShape", Quantize2BitShape)
+.set_attr<nnvm::FInferType>("FInferType", Quantize2BitType)
+.set_attr<FCompute>("FCompute<cpu>", Quantize2BitCompute<cpu>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseNone{"_quantize_2bit"})
+.set_attr<nnvm::FMutateInputs>("FMutateInputs",
+[](const nnvm::NodeAttrs& attrs) {
+    return std::vector<uint32_t>{1, 2};
+})
+.add_argument("gradient_array", "NDArray-or-Symbol", "A ndarray/symbol of type `float32`")
+.add_argument("residual_array", "NDArray-or-Symbol", "A ndarray/symbol of type `float32`")
+.add_argument("quantized_array", "NDArray-or-Symbol", "A ndarray/symbol of type `float32`")
+.add_arguments(TwoBitParam::__FIELDS__());
+
+NNVM_REGISTER_OP(_contrib_create_2bit)
```

Comment:
Dont like this. The name is inconsistent. Plus this shouldn't be exposed as an operator

Reply:
Hmm, I'll just remove all these operators. We don't even need this function then